controllers/packetmachine_controller.go

```diff
@@ -385,13 +387,16 @@ func (r *PacketMachineReconciler) reconcile(ctx context.Context, machineScope *s
 
 		if machineScope.PacketCluster.Spec.VIPManager == "CPEM" {
 			controlPlaneEndpoint, _ = r.PacketClient.GetIPByClusterIdentifier(
+				ctx,
 				machineScope.Cluster.Namespace,
 				machineScope.Cluster.Name,
 				machineScope.PacketCluster.Spec.ProjectID)
 			if len(controlPlaneEndpoint.Assignments) == 0 && machineScope.IsControlPlane() {
-				if _, _, err := r.PacketClient.DeviceIPs.Assign(dev.ID, &packngo.AddressStruct{
-					Address: controlPlaneEndpoint.Address,
-				}); err != nil {
+				apiRequest := r.PacketClient.DevicesApi.CreateIPAssignment(nil, *dev.Id)
```

Comment:
```suggestion
				apiRequest := r.PacketClient.DevicesApi.CreateIPAssignment(ctx, *dev.Id)
```

Reply:
and `dev.GetId()` if you want to take that route.  The unchecked pointers have me on edge, but as called out in elsewhere in the PR nil exceptions are more obvious to track down than passed along empty strings that get silently accepted.